sql/src/main/java/io/crate/expression/tablefunctions/GenerateSeries.java

```diff
@@ -71,69 +65,104 @@
 public final class GenerateSeries<T extends Number> extends TableFunctionImplementation<T> {
 
     public static final FunctionName NAME = new FunctionName(PgCatalogSchemaInfo.NAME, "generate_series");
-    private final FunctionInfo info;
-    private final T defaultStep;
-    private final BinaryOperator<T> minus;
-    private final BinaryOperator<T> plus;
-    private final BinaryOperator<T> divide;
-    private final Comparator<T> comparator;
-    private final RowType returnType;
 
     public static void register(TableFunctionModule module) {
-        Param startAndEnd = Param.of(DataTypes.LONG, DataTypes.INTEGER, DataTypes.TIMESTAMPZ, DataTypes.TIMESTAMP);
-        Param stepType = Param.of(DataTypes.LONG, DataTypes.INTEGER, DataTypes.INTERVAL);
-        FuncParams.Builder paramsBuilder = FuncParams
-            .builder(startAndEnd, startAndEnd)
-            .withVarArgs(stepType)
-            .limitVarArgOccurrences(1);
-        module.register(NAME, new BaseFunctionResolver(paramsBuilder.build()) {
-            @Override
-            public FunctionImplementation getForTypes(List<DataType> types) throws IllegalArgumentException {
-                DataType<?> startType = types.get(0);
-                DataType<?> stopType = types.get(1);
-                assert startType.equals(stopType) : "Start and stop type must be the same, got: " + startType + " and " + stopType;
-                if (types.size() == 2 && !startType.equals(DataTypes.INTEGER) && !startType.equals(DataTypes.LONG)) {
+        // without step
+        module.register(
+            Signature.table(
+                NAME,
+                DataTypes.LONG.getTypeSignature(),
+                DataTypes.LONG.getTypeSignature(),
+                new RowType(List.of(DataTypes.LONG)).getTypeSignature()),
+            (signature, argTypes) -> new GenerateSeries<>(
+                signature, argTypes, 1L, (x, y) -> x - y, Long::sum, (x, y) -> x / y, Long::compare)
```

Comment:
minor: I'd prefer having each argument in a dedicated line for improved readability (also related to the other registrations below).